src/WebRTC/SessionDescriptionHandler.js

```diff
@@ -0,0 +1,414 @@
+"use strict";
+/**
+ * @fileoverview SessionDescriptionHandler
+ */
+
+ /* SessionDescriptionHandler
+  * @class PeerConnection helper Class.
+  * @param {SIP.Session} session
+  * @param {Object} [options]
+  */
+module.exports = function (SIP) {
+
+// Constructor
+var SessionDescriptionHandler = function(session, options) {
+  this.options = options || {};
+
+  this.logger = session.ua.getLogger('sip.invitecontext.sessionDescriptionHandler', session.id);
+  this.session = session;
+
+  this.CONTENT_TYPE = 'application/sdp';
+
+  // TODO: Move to UA options
+  this.modifiers = this.options.modifiers || [];
+  if (!Array.isArray(this.modifiers)) {
+    this.modifiers = [this.modifiers];
+  }
+
+  // TODO: Other sdp modifiers on the UA
+  if (this.session.ua.configuration.hackStripTcp) {
+    this.modifiers.push(function(sdp) {
+      sdp = sdp.replace(/^a=candidate:\d+ \d+ tcp .*?\r\n/img, "");
+      return SIP.Utils.Promise.resolve(sdp);
+    });
+  }
+
+  var environment = global.window || global;
+  this.WebRTC = {
+    MediaStream           : environment.MediaStream,
+    getUserMedia          : environment.navigator.mediaDevices.getUserMedia.bind(environment.navigator.mediaDevices),
+    RTCPeerConnection     : environment.RTCPeerConnection,
+    RTCSessionDescription : environment.RTCSessionDescription
+  };
+
+  var servers = this.prepareIceServers(this.options.stunServers, this.options.turnServers);
+  this.RTCConstraints = this.options.RTCConstraints || {};
+
+  this.initPeerConnection(servers);
+};
+
+/**
+ * @param {SIP.Session} session
+ * @param {Object} [options]
+ */
+
+ // TODO: The options are going to disappear, and the handler will be responsible for pulling it's own options from the UA
+SessionDescriptionHandler.defaultFactory = function defaultFactory (session, options) {
+  return new SessionDescriptionHandler(session, options);
+};
+
+SessionDescriptionHandler.prototype = Object.create(SIP.SessionDescriptionHandler.prototype, {
+  // Funtions the sesssion can use
+  // TODO: Comments
+  // TODO:
+  close: {writable: true, value: function () {
+    return;
+  }},
+
+  /**
+   *
+   */
+  getDescription: {writable: true, value: function (constraints, modifiers) {
+    var self = this;
+
+    // We will override the saved constraints with new constraints if provided, otherwise use saved constraints
+    this.constraints = constraints || this.constraints || {};
```

Comment:
This could mess up the underlying WebRTC library if the `constraints` argument passed to `getDescription` is an empty object.